src/engine.ts

```diff
@@ -833,8 +830,7 @@ export default class Engine {
     //mark as activated special action for this turn
     this.player(player).activateEvents(specialEvent, true);
 
-    if ( income === "=>range" || income === "=>step" ){
-      this.buildMinePhase(player, income);
+    if ( income == Resource.TemporaryRange || income == Resource.TemporaryStep ){
```

Comment:
this can still be removed altogether. The `this.endTurnPhase` won't add anything if there's already a subcommand for the player.

Reply:
well, a special action has to have an `endTurnPhase`. If the special action is buildMine, it will take care of the endTurnPhase. Not sure that every special action will manage its own endTurnPhase. Anyways, I'll removed it for now

Reply:
If there is a required action, then end turn phase will wait until the subcommand is done. You can test in the UI, you should only see the "build a mine" command & not the free actions.

Reply:
```
endTurnPhase(player: PlayerEnum, fromCommand: Command){
    // ...
    if (this.nextSubcommandPlayer() === player) {
      return;
    }
}
```

This makes calling `endTurnPahse` the same as doing nothing when there's already a subcommand (like building a mine) in the queue

Reply:
understood